Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInputEditor.cs

```diff
@@ -110,16 +111,24 @@ public override void OnInspectorGUI()
             DoHelpCreateAssetUI();
 
             // UI config section.
-            /*
-            var uiProperty = serializedObject.FindProperty("m_UIEventSystem");
+            var uiModuleProperty = serializedObject.FindProperty("m_UIInputModule");
             if (m_UIPropertyText == null)
-                m_UIPropertyText = EditorGUIUtility.TrTextContent("UI", uiProperty.tooltip);
+                m_UIPropertyText = EditorGUIUtility.TrTextContent("UI Input Module", uiModuleProperty.tooltip);
             EditorGUI.BeginChangeCheck();
-            EditorGUILayout.PropertyField(uiProperty, m_UIPropertyText);
-            if (EditorGUI.EndChangeCheck() || !m_UIConnectionInitialized)
-                OnUIConnectionChange();
-            DoHelpSetUpUnityUI();
-            */
+            EditorGUILayout.PropertyField(uiModuleProperty, m_UIPropertyText);
+            if (EditorGUI.EndChangeCheck())
+                serializedObject.ApplyModifiedProperties();
+
+            if (uiModuleProperty.objectReferenceValue != null)
+            {
+                var uiModule = uiModuleProperty.objectReferenceValue as InputSystemUIInputModule;
+                if (actionsProperty.objectReferenceValue != null && uiModule.actionsAsset != actionsProperty.objectReferenceValue)
+                {
+                    EditorGUILayout.HelpBox("The referenced InputSystemUIInputModule is configured using differnet input actions then this PlayerInput. They should match if you want to synch PlayerInput actions to the UI input.", MessageType.Warning);
+                    if (GUILayout.Button(m_FixInputModuleText))
+                        InputSystemUIInputModuleEditor.ReassignActions(uiModule, actionsProperty.objectReferenceValue as InputActionAsset);
+                }
+            }
```

Comment:
Think it would be great to also detect the case here where the UI event system isn't set up for the new input system at all.

Reply:
I don't understand. _If_ you have an `InputSystemUIInputModule` reference assigned here, you also have the UI Event system set up for the new input system. If you don't, then you are not connecting PlayerInput to any UI, so what's the point?

Reply:
The workflow I have in mind here is something like

1) User adds a UI canvas and some UI elements. This implicitly adds an EventSystem configured for the old input system.
2) User adds a PlayerInput.

Now the user has a conflicting setup but without going to the implicitly added EventSystem and reconfiguring the system, the user won't know.

Think it would be great when in step 2 PlayerInput immediately tells the user "you know, there's something I see that I know won't work. want me to fix it?"

Reply:
As we discussed, I'll add a general warning for using "old input" when it is not enabled.